### PR TITLE
Adding CI e2e for testing GPU support in Kubernetes

### PR DESCRIPTION
**Pull request by vishh**
For enabling tests added by https://github.com/kubernetes/kubernetes/pull/45136

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -2224,6 +2224,13 @@
       "--mode=local"
     ], 
     "scenario": "kubernetes_e2e"
+  },
+  "ci-kubernetes-e2e-gce-gpu": {
+    "args": [
+      "--env-file=platforms/gce.env",
+      "--env-file=jobs/ci-kubernetes-e2e-gce-gpu.env"
```
Comment (krzyzacy) [inline jobs/config.json]: rename the actual env file to `ci-kubernetes-e2e-gce-gpu.env` :-) that's what is ranting

Comment (vishh) [inline jobs/config.json]: oops. Fixed it.

File: jobs/config.json [hunk 2]
```diff
@@ -547,6 +547,13 @@
     ], 
     "scenario": "kubernetes_e2e"
   }, 
+  "ci-kubernetes-e2e-gce-gpu": {
+    "args": [
+      "--env-file=platforms/gce.env", 
+      "--env-file=jobs/ci-kubernetes-e2e-gce-gpu.env"
```
Comment (krzyzacy) [inline jobs/config.json]: oh sorry missed this, you'll need `--mode=local` for prow jobs since prow does not support docker. I'll add a unit-test.

**Pull request merged**